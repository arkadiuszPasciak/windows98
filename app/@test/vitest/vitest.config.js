import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"
import tsconfigPaths from "vite-tsconfig-paths"

export default mergeConfig(
	sharedConfig,
	defineProject({
		plugins: [tsconfigPaths({ loose: true })],
	})
)
