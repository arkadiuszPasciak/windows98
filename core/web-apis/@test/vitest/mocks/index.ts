export * from "./canvas-api.mock"
export * from "./media-devices-api.mock"
export * from "./screen-api.mock"
export * from "./video-api.mock"
