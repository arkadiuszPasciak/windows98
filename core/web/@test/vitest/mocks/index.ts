export * from "./battery-status-api.mock"
export * from "./canvas-api.mock"
export * from "./clipboard-api.mock"
export * from "./cookie-store-api.mock"
export * from "./device-memory-api.mock"
export * from "./fullscreen-api.mock"
export * from "./location-api.mock"
export * from "./media-devices-api.mock"
export * from "./screen-api.mock"
export * from "./user-agent-client-hints-api.mock"
export * from "./video-api.mock"
export * from "./web-storage.mock"
