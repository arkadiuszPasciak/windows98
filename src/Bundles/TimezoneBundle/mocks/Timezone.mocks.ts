import { ITimezones } from '@Bundles/TimezoneBundle/supports/Timezone.supports'

export const timezones = [
  {
    value: -12,
    name: 'TimezoneBundle.time-12',
  },
  {
    value: -11,
    name: 'TimezoneBundle.time-11',
  },
  {
    value: -10,
    name: 'TimezoneBundle.time-10',
  },
  {
    value: -9,
    name: 'TimezoneBundle.time-9',
  },
  {
    value: -8,
    name: 'TimezoneBundle.time-8',
  },
  {
    value: -7,
    name: 'TimezoneBundle.time-7',
  },
  {
    value: -6,
    name: 'TimezoneBundle.time-6',
  },
  {
    value: -5,
    name: 'TimezoneBundle.time-5',
  },
  {
    value: -4,
    name: 'TimezoneBundle.time-4',
  },
  {
    value: -3,
    name: 'TimezoneBundle.time-3',
  },
  {
    value: -2,
    name: 'TimezoneBundle.time-2',
  },
  {
    value: -1,
    name: 'TimezoneBundle.time-1',
  },
  {
    value: 0,
    name: 'TimezoneBundle.time+0',
  },
  {
    value: 1,
    name: 'TimezoneBundle.time+1',
  },
  {
    value: 2,
    name: 'TimezoneBundle.time+2',
  },
  {
    value: 3,
    name: 'TimezoneBundle.time+3',
  },
  {
    value: 4,
    name: 'TimezoneBundle.time+4',
  },
  {
    value: 5,
    name: 'TimezoneBundle.time+5',
  },
  {
    value: 6,
    name: 'TimezoneBundle.time+6',
  },
  {
    value: 7,
    name: 'TimezoneBundle.time+7',
  },
  {
    value: 8,
    name: 'TimezoneBundle.time+8',
  },
  {
    value: 9,
    name: 'TimezoneBundle.time+9',
  },
  {
    value: 10,
    name: 'TimezoneBundle.time+10',
  },
  {
    value: 11,
    name: 'TimezoneBundle.time+11',
  },
  {
    value: 12,
    name: 'TimezoneBundle.time+12',
  },
] as Array<ITimezones>
