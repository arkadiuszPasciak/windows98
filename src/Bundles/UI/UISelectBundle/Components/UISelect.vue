<template>
  <div
    class="UISelect"
    :class="props.labelName ? `label-position-${labelPosition}` : ''"
  >
    <label
      v-if="labelName"
      class="label"
      :data-test="`ui-select-label-${id}`"
      :for="id"
    >
      {{ labelName }}
    </label>
    <select
      :id="id"
      v-model="value"
      class="select"
      :name="id"
      :data-test="`ui-select-${id}`"
      :disabled="disabled"
      :required="required"
    >
      <option
        v-for="option in options"
        :key="`${id}-${option}`"
        :value="option.value"
        class="option"
      >
        {{ t(option.name) }}
      </option>
    </select>
  </div>
</template>

<script setup lang="ts">
  import { computed, PropType } from 'vue'
  import { useI18n } from 'vue-i18n'
  import { TUISelectPropsObject } from '@Bundles/UISelectBundle/Supports/UISelect.supports'

  const props = defineProps({
    id: {
      type: String,
      required: true,
    },
    required: {
      type: Boolean,
      default: false,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    modelValue: {
      type: [String, Number],
      required: true,
    },
    options: {
      type: Object as PropType<TUISelectPropsObject>,
      required: true,
    },
    labelName: {
      type: String,
      required: false,
    },
    labelPosition: {
      type: String as PropType<'top' | 'left'>,
      default: 'top',
    },
  })

  const { t } = useI18n()

  const emit = defineEmits(['update:modelValue'])

  const value = computed({
    get() {
      return props.modelValue
    },
    set(value) {
      emit('update:modelValue', value)
    },
  })
</script>

<i18n src="@Bundles/LanguageBundle/Locales/Language.locales.json"></i18n>
<i18n src="@Bundles/ThemeBundle/Locales/Theme.locales.json"></i18n>
<i18n src="@Bundles/CalendarBundle/Locales/Calendar.locales.json"></i18n>
<i18n src="@Bundles/TimezoneBundle/Locales/Timezone.locales.json"></i18n>
<i18n src="@Bundles/FileBundle/Locales/File.locales.json"></i18n>

<style
  lang="scss"
  scoped
  src="@Bundles/UISelectBundle/Styles/UISelect.styles.scss"
/>
