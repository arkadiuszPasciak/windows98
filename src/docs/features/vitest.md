# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
