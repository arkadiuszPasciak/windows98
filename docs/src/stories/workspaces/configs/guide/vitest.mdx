import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

