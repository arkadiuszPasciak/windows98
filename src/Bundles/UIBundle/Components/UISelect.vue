<template>
  <select
    :id="id"
    v-model="value"
    class="UISelect"
    :class="`variant-${variant}`"
    :name="id"
    :disabled="disabled"
    :required="required"
  >
    <option
      v-for="option in options"
      :key="`${id}-${option}`"
      :value="option.value"
      class="option"
    >
      {{ option.name }}
    </option>
  </select>
</template>

<script setup lang="ts">
  import { computed, PropType } from 'vue'
  import { TUISelectPropsObject } from '@Bundles/UIBundle/Supports/UISelect.supports'

  const props = defineProps({
    variant: {
      type: String as PropType<'primary'>,
      defaut: 'primary',
    },
    id: {
      type: String,
      required: true,
    },
    required: {
      type: Boolean,
      default: false,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    modelValue: {
      type: String,
      required: true,
    },
    options: {
      type: Object as PropType<TUISelectPropsObject>,
      required: true,
    },
  })

  const emit = defineEmits(['update:modelValue'])

  const value = computed({
    get() {
      return props.modelValue
    },
    set(value) {
      emit('update:modelValue', value)
    },
  })
</script>

<style lang="scss" scoped>
  @import '@Bundles/UIBundle/Styles/UISelect.styles.scss';
</style>
