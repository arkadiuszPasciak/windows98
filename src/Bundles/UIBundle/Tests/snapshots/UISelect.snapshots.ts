import { mount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import UISelect from '@Bundles/UIBundle/Components/UISelect.vue'
import { languagesOptions } from '@Bundles/LanguageBundle/Mocks/Language.mocks'

const defaultWrapper = mount(UISelect, {
  props: {
    id: 'ui-select-test',
    options: languagesOptions,
    modelValue: languagesOptions[0].value,
  },
})

describe('[UIBundle]<Snapshots>(UISelect)', async () => {
  it('renders correctly component', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
})
