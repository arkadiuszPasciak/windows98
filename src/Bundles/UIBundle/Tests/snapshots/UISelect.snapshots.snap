// Vitest Snapshot v1

exports[`[UIBundle]<Snapshots>(UISelect) > renders correctly component 1`] = `
<select
  class="UISelect variant-undefined"
  data-v-d3c68ad1=""
  id="ui-select-test"
  name="ui-select-test"
>
  
  <option
    class="option"
    data-v-d3c68ad1=""
    value="en"
  >
    Language.english
  </option>
  <option
    class="option"
    data-v-d3c68ad1=""
    value="pl"
  >
    Language.polish
  </option>
  
</select>
`;
